src/main/kotlin/org/ageseries/libage/sim/electrical/mna/Circuit.kt

```diff
@@ -663,7 +663,7 @@ class Circuit {
                     1 -> ":e"
                     else -> ""
                 }
-                val node = it.representative as Node
+                val node = (it.representative as Pin).node ?: throw Exception("This pin has a null Node!")
```

Comment:
You might want Kotlin's `error` instead of `throw Exception` here; it'll make Codacy a little happier :)